crates/runtime/src/datafusion.rs

```diff
@@ -540,11 +546,19 @@ impl DataFusion {
 
         accelerated_table_builder.zero_results_action(acceleration_settings.on_zero_results);
 
-        accelerated_table_builder.cache_provider(self.cache_provider.clone());
+        accelerated_table_builder.cache_provider(self.cache_provider());
 
         Ok(accelerated_table_builder.build().await)
     }
 
+    pub fn cache_provider(&self) -> Option<Arc<QueryResultsCacheProvider>> {
+        let Ok(provider) = self.cache_provider.read() else {
```

Comment:
@y-f-u  - can this impact performance as we are adding lock that we need to acquire during each query

Reply:
yeah, good point. it will have some impact as it is still atomic.

Reply:
I did a quick test on the read() performance. (machine spec may differ this) it takes about 60ms for 1 million operation.

Reply:
also this is not going to worsen the performance comparing to the previous release, as we removed the df.read() which is essentially the same.